prime-router/src/main/kotlin/cli/SenderUtilsCommand.kt

```diff
@@ -0,0 +1,158 @@
+package gov.cdc.prime.router.cli
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.PrintMessage
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.choice
+import gov.cdc.prime.router.FileSettings
+import gov.cdc.prime.router.Sender
+import gov.cdc.prime.router.azure.HttpUtilities
+import gov.cdc.prime.router.azure.SenderAPI
+import gov.cdc.prime.router.tokens.Scope
+import gov.cdc.prime.router.tokens.SenderUtils
+import gov.cdc.prime.router.tokens.TokenAuthentication
+import java.io.File
+
+class AddPublicKey : SingleSettingCommand(
+    name = "addkey",
+    help = """
+    Add a public key to an existing sender's setting
+
+    Prior to calling this, generate Elliptic Curve private + public key pair for use with ES384 signatures:
+      openssl ecparam -genkey -name secp384r1 -noout -out my-es-keypair.pem
+      openssl ec -in my-es-keypair.pem -pubout -out my-es-public-key.pem
+
+    Or, generate RSA keyusing
+      openssl genrsa -out my-rsa-keypair.pem 2048
+            openssl rsa -in my-rsa-keypair.pem -outform PEM -pubout -out my-rsa-public-key.pem
+    """.trimIndent(),
+    settingType = SettingType.SENDER,
+    operation = Operation.GET
+) {
+    val publicKeyFilename by option("--public-key",
+        help = "Path to public key .pem file",
+        metavar = "<public-pem-key-file>")
+        .required()
```

Comment:
clikt has a file() option that might be helpful to remove some of the file logic in this method. See an example here: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/cli/CredentialsCli.kt#L89